t_frang/test_config.py

```diff
@@ -0,0 +1,799 @@
+                    http_chain {-> vhost_1;}
+                """,
+            ),
+            param(
+                name="vhost_to_location",
+                config="""
+                    vhost vhost_1 {
+                        proxy_pass default;
+                        frang_limits {http_methods get post put;}
+                        location prefix "/vhost_1" {
+                            cache_fulfill * *; 
+                            proxy_pass default;
+                            frang_limits {http_methods post put;}
+                        }
+                    }
+                    http_chain {-> vhost_1;}
+                """,
+            ),
+        ]
+    )
+    @unlimited_rate_on_tempesta_node
+    def test_overriding(self, name, config: str):
+        """
+        Tempesta has different frang limits in global/vhost/location.
+        """
+        self.__update_tempesta_config(config)
+        self.start_all_services()
+        client = self.get_client("deproxy")
+
+        client.send_request(client.create_request(method="GET", uri="/vhost_1", headers=[]), "403")
+        self.assertTrue(self.oops.find("frang: restricted HTTP method for"))
+
+    def _test_not_override_http_methods(self):
+        client = self.get_client("deproxy")
+        client.send_request(client.create_request(method="GET", headers=[]), "403")
+        self.assertTrue(self.oops.find("frang: restricted HTTP method for"))
+
+    def _test_not_override_concurrent_tcp_connections(self):
+        client = self.get_client("deproxy")
+        client_1 = self.get_client("deproxy_1")
+
+        client.make_request(client.create_request(method="GET", headers=[]))
+        client_1.make_request(client.create_request(method="GET", headers=[]))
+
+        client.wait_for_response(timeout=2)
+        client_1.wait_for_response(timeout=2)
+
+        time.sleep(2)
+
+        self.assertTrue(self.oops.find("frang: connections max num. exceeded for"))
+        self.assertEqual(1, len(client.responses) + len(client_1.responses))
+
+    def _test_not_override_http_body_len_0(self):
+        client = self.get_client("deproxy_http")
+        request = (
+            f"POST /1234 HTTP/1.1\r\nHost: localhost\r\nContent-Length: 1000\r\n\r\n{'x' * 1000}"
+        )
+        client.send_request(request, "200")
+        self.assertFalse(self.oops.find("frang: HTTP body length exceeded for"))
+
+    def _test_not_override_http_body_len_1(self):
+        client = self.get_client("deproxy_http")
+        request = f"POST /1234 HTTP/1.1\r\nHost: localhost\r\nContent-Length: 2\r\n\r\n{'x' * 2}"
+        client.send_request(request, "403")
+        self.assertTrue(self.oops.find("frang: HTTP body length exceeded for"))
+
+    def _test_not_override_http_body_len_2(self):
+        client = self.get_client("deproxy_http")
+        request = f"POST /1234 HTTP/1.1\r\nHost: localhost\r\nContent-Length: 2\r\n\r\n{'x' * 2}"
+        client.send_request(request, "403")
```

Comment:
Seems here we need `client.send_request(request, "200")` because http_body_len is explicitly overridden to 3